### PR TITLE
Created medfilt

### PR DESCRIPTION
**Pull request by ClaudioStrumia**
Created medfilt from scipy.signal

### CODE REVIEW

**Comment by atrabattoni** [comment]
It would be nice if `kernel_size` was passed as a dict. The dict keys would be the dim over which to apply a median filtering. The related values would be the size of the kernel along that direction. If not all dims are provided, then fill with one the kernel size along the missing dims.

File: medfilt.py
Comment (atrabattoni) [inline medfilt.py:1]: You should remove this file

File: tests/codes/test_medfilt.py
Comment (atrabattoni) [inline tests/codes/test_medfilt.py:1]: You should remove this file

File: xdas/signal.py [hunk 1]
```diff
@@ -227,3 +227,50 @@ def sliding_mean_removal(da, wlen, window="hann", pad_mode="reflect", dim="dista
...
+    and a size of 3 for the median filtering along the space dimension.
+    The database is synthetic data.
+    >>> from xdas.synthetics import generate
+    >>> da = generate()
+    >>> dimensions = np.array([coord for coord in da.coords])
+    >>> kernel_length = [7, 3]
+    >>> dims = dict(zip(dimensions, kernel_length))
+    >>> filtered_da = medfilt(da, dims)
+    """
+    coordinates = np.array([coord for coord in da.coords])
```
Comment (atrabattoni) [inline xdas/signal.py:265]: you do weird stuff. I would somthing like:

```python
kernel_size = tuple(kernel_size[dim] if dim in kernel_size else 1 for dim in db.dims)
```

File: tests/test_signal.py [hunk 2]
```diff
@@ -62,3 +62,12 @@ def test_sliding_window_removal(self):
         da = xr.DataArray(data, {"distance": s})
         da = xp.sliding_mean_removal(da, 0.1 * n * d)
         assert np.allclose(da, 0)
+
+    def test_medfilt(self):
```
Comment (atrabattoni) [inline tests/test_signal.py:66]: Maybe just test to see what happends if you do not provide all dims.

File: xdas/signal.py [hunk 3]
```diff
@@ -227,3 +227,50 @@ def sliding_mean_removal(da, wlen, window="hann", pad_mode="reflect", dim="dista
...
+    by selecting a size of 7 for the median filtering along the time dimension
+    and a size of 3 for the median filtering along the space dimension.
+    The database is synthetic data.
+    >>> from xdas.synthetics import generate
+    >>> da = generate()
+    >>> dimensions = np.array([coord for coord in da.coords])
+    >>> kernel_length = [7, 3]
+    >>> dims = dict(zip(dimensions, kernel_length))
+    >>> filtered_da = medfilt(da, dims)
+    """
```
Comment (atrabattoni) [inline xdas/signal.py:264]: You could check that all provide dims are indeed inside the database

```python
if not all(dim in db.dims for dim in kernel_size):
    raise ValueError("dims provided not in database")
```

**Review by atrabattoni — CHANGES_REQUESTED** [review]
Your are almost there. Just some more checks and simpler code.

File: xdas/signal.py (hunk 1 above)
Comment (ClaudioStrumia) [reply xdas/signal.py]: Works like a charm.. thanks

File: xdas/signal.py (hunk 3 above)
Comment (ClaudioStrumia) [reply xdas/signal.py:264]: Done, thanks.

File: tests/test_signal.py (hunk 2 above)
Comment (atrabattoni) [reply tests/test_signal.py:66]: You need to merge the dev branch in your working branch to get the latest updates

**Pull request merged**